GHzDACs/Cleanup/dac.py

```diff
@@ -936,8 +1024,534 @@ class DacRunner_Build11(DacRunner_Build7):
+        runner = self.RUNNER_CLASS(self, reps, start_delay, jt_entries, jt_counters, sram, loop_delay=loop_delay)
+        return runner
+
+    # let's check the various registry functions.
+    # OK means old implementation should work
+
+    @classmethod
+    def regPing(cls):
+        """Returns a numpy array of register bytes to ping DAC register"""
+        regs = np.zeros(cls.REG_PACKET_LEN, dtype='<u1')
+        regs[0] = 1  # Master mode to prevent daisy passthrough.
+        regs[1] = 1  # Readback after 2us
+        regs[51] = MONITOR_0
+        regs[52] = MONITOR_1
+        return regs
+
+    @classmethod
+    def regPllQuery(cls):
+        """Returns a numpy array of register bytes to query PLL status"""
+        return cls.regPing()
+
+    @classmethod
+    def regSerial(cls, op, data):
+        regs = np.zeros(cls.REG_PACKET_LEN, dtype='<u1')
+        regs[0] = 1  # Master mode to prevent daisy passthrough
+        regs[1] = 1  # Readback = readback after 2us to allow for serial
+        regs[47] = op  # Set serial operation mode to op
+        regs[48:51] = littleEndian(data, 3)  # Serial data
+        regs[51] = MONITOR_0
+        regs[52] = MONITOR_1
+        return regs
+
+    @classmethod
+    def regPllReset(cls):
+        """Send reset pulse to 1GHz PLL"""
+        regs = np.zeros(cls.REG_PACKET_LEN, dtype='<u1')
+        regs[0] = 1  # master mode to prevent daisy passthrough
+        regs[1] = 1
+        regs[46] = 0x80  # Set d[7..0] to 10000000 = reset 1GHz PLL pulse
+        return regs
+
+    @classmethod
+    def regRun(cls, reps, page, slave, delay, blockDelay=None, sync=249,
+               loop_delay=0, readback=True, monitor_0=None, monitor_1=None):
+        # TODO: probably get rid of page, blockDelay
+        if blockDelay is not None:
+            raise ValueError("JT board got a non-None blockDelay: ", blockDelay)
+        if page:
+            raise ValueError("JT board got a non-zero page: ", page)
+        regs = np.zeros(cls.REG_PACKET_LEN, dtype='<u1')
+        # old version of slave: 0 = master, 1 = slave, 3 = idle (bit 43)
+        # new version: 0 = idle, 1 = master, 2 = test, 3 = slave (bit 0)
+        if slave == 0:
+            start = 1
+        elif slave == 1:
+            start = 3
+        elif slave == 3:
+            start = 0
+        else:
+            raise ValueError('"slave" must be 0, 1, or 3, not %s' % slave)
+        regs[0] = start
+        regs[1] = int(readback)
+        regs[13:15] = littleEndian(reps, 2)
+        regs[15:17] = littleEndian(loop_delay, 2)
+        regs[43:45] = littleEndian(int(delay), 2)
+        regs[45] = sync
+        regs[17] = 0  # Which jump table to count activations of
+        regs[51] = monitor_0 if monitor_0 is not None else MONITOR_0
+        regs[52] = monitor_1 if monitor_1 is not None else MONITOR_1
+
+        return regs
+
+    @classmethod
+    def regRunSimple(cls, readback=True):
+        """
+        just run the thing
```

Comment:
There were enough cases where I needed to get a packet to just run it once to play the SRAM, and it seemed simpler to do it this way than to always remember reps=1, page=0, slave=0, delay=0. I can do a find/replace to remove this function if you like.